File: src/test/java/akostenko/aicars/math/DecartTest.java
package akostenko.aicars.math;

import static java.lang.StrictMath.PI;
import static java.lang.StrictMath.sqrt;
import static org.junit.Assert.*;

import org.junit.Test;

public class DecartTest {

    @Test
    public void toPolarTest_OnAxis() throws Exception {
        Polar right = new Decart(1, 0).toPolar();
        assertEquals(new Polar(1, 0), right);

        Polar up = new Decart(0, 1).toPolar();
        assertEquals(new Polar(1, PI/2), up);

        Polar left = new Decart(-1, 0).toPolar();
        assertEquals(new Polar(1, PI), left);

        Polar down = new Decart(0, -1).toPolar();
        assertEquals(new Polar(1, 3*PI/2), down);
    }

    @Test
    public void toPolarTest_InQuartiles() throws Exception {
        Polar northEast = new Decart(1, 1).toPolar();
        assertEquals(new Polar(sqrt(2), PI/4), northEast);

        Polar northWest = new Decart(-1, 1).toPolar();
        assertEquals(new Polar(sqrt(2), 3*PI/4), northWest);

        Polar southWest = new Decart(-1, -1).toPolar();
        assertEquals(new Polar(sqrt(2), 5*PI/4), southWest);

        Polar southEast = new Decart(1, -1).toPolar();
        assertEquals(new Polar(sqrt(2), 7*PI/4), southEast);
    }

}

File: src/test/java/akostenko/aicars/math/DecartTest.kt
package akostenko.aicars.math

import java.lang.StrictMath.PI
import java.lang.StrictMath.sqrt
import org.junit.Assert.*

import org.junit.Test

class DecartTest {

    @Test
    @Throws(Exception::class)
    fun toPolarTest_OnAxis() {
        val right = Decart(1.0, 0.0).toPolar()
        assertEquals(Polar(1.0, 0.0), right)

        val up = Decart(0.0, 1.0).toPolar()
        assertEquals(Polar(1.0, PI / 2), up)

        val left = Decart(-1.0, 0.0).toPolar()
        assertEquals(Polar(1.0, PI), left)

        val down = Decart(0.0, -1.0).toPolar()
        assertEquals(Polar(1.0, 3 * PI / 2), down)
    }

    @Test
    @Throws(Exception::class)
    fun toPolarTest_InQuartiles() {
        val northEast = Decart(1.0, 1.0).toPolar()
        assertEquals(Polar(sqrt(2.0), PI / 4), northEast)

        val northWest = Decart(-1.0, 1.0).toPolar()
        assertEquals(Polar(sqrt(2.0), 3 * PI / 4), northWest)

        val southWest = Decart(-1.0, -1.0).toPolar()
        assertEquals(Polar(sqrt(2.0), 5 * PI / 4), southWest)

        val southEast = Decart(1.0, -1.0).toPolar()
        assertEquals(Polar(sqrt(2.0), 7 * PI / 4), southEast)
    }

}